community/kernel/src/main/java/org/neo4j/kernel/impl/query/Neo4jTransactionalContext.java

```diff
@@ -109,20 +139,34 @@ public void close( boolean success )
     @Override
     public void commitAndRestartTx()
     {
-        transaction.success();
-        transaction.close();
+        QueryRegistryOperations oldQueryRegistryOperations = statement.queryRegistration();
```

Comment:
something is going on here...we need at least javadoc with explanation what we doing here and more then one test if possible